platform/cog-platform-fdo.c

```diff
@@ -722,7 +817,6 @@ keyboard_on_enter (void *data,
                    struct wl_surface *surface,
                    struct wl_array *keys)
 {
-    g_assert (surface == win_data.wl_surface);
```

Comment:
Does this assertion need to be removed?

Reply:
I had to remove it during the development cycle but checking again now, the assertion is no longer hit, so let's keep it :)